source/common/upstream/upstream_impl.h

```diff
@@ -757,6 +758,15 @@ class ClusterInfoImpl : public ClusterInfo,
   std::vector<Http::Protocol>
   upstreamHttpProtocol(absl::optional<Http::Protocol> downstream_protocol) const override;
 
+  // Http::FilterChainFactory
+  void createFilterChain(Http::FilterChainManager& manager) const override {
+    Http::FilterChainUtility::createFilterChainForFactories(manager, http_filter_factories_);
+  }
+  bool createUpgradeFilterChain(absl::string_view, const UpgradeMap*,
+                                Http::FilterChainManager&) const override {
+    return false;
```

Comment:
Mind adding a comment? Presumably something like "// Upgrade filters are not supported on upstream connections"?

Reply:
1) wasn't needed at the time and 2) when it's added we'll have to be careful to document so folks don't get confused.  We can apply filters to upgrades where we're still doing L7 (passthrough upgrades) but not where we're handling the upgrade (terminating upgrades) because the upstream is then a TCP upstream and only handles L4 filters.

Reply:
for upgrades you can have something like
[connect headers][non-http-body payload]
Envoy can either proxy this through as a connect request, forwarding on both the headers and payload, or terminate the connect request, strip the headers and forward only the payload on.  The former is proxied as HTTP, so gets upstream filters.  the latter is forwarded as TCP so no upstream L7 filters

see https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/http/upgrades for more details

Reply:
Ahh I see what you're saying, in my case I'm specifically working with `WebSocket HTTP/1.1` upgrades so I should be fine?

Reply:
yeah I think the filters would apply but the docs still need to be clear for folks with other use cases